components/MapComponent.tsx

```diff
@@ -77,28 +91,73 @@ export function MapComponent<T extends mapMarker> ({
     ))
   }
 
-  // When clicking on the marker, zoom towards where it is.
-  function onMarkerClick (lat: number, long: number) {
-    mapRef.current.animateToRegion({
-      latitude: lat,
-      longitude: long,
-      latitudeDelta: initialRegion.latitudeDelta - 0.02,
-      longitudeDelta: initialRegion.longitudeDelta - 0.02
+  // Check to see if the circle is for the right selected marker.
+  const checkCircleKey = (point) => {
+    if (point.markedMarker.key === selectedMarker) {
+      return (
+        <Circle
+          center={{
+            latitude: point.markedMarker.location.latitude,
+            longitude: point.markedMarker.location.longitude
+          }}
+          radius={initialRadius.radius}
+          fillColor={customizationCreation(point.markedMarker).circleFillColor}
+          strokeColor={
+            customizationCreation(point.markedMarker).circleStrokeColor
+          }
+          strokeWidth={
+            customizationCreation(point.markedMarker).circleStrokeWidth
+          }
+        />
+      )
+    }
+  }
+
+  const circleCreations = () => {
+    return markerData.map((point) => {
+      const createdCircle = checkCircleKey(point)
+      return createdCircle
     })
   }
 
-  // When tapping "Re-Center" button, move to the marker in question.
-  /* function onRecenter () {
-    mapRef.current.animateToRegion({
-      latitude: initialRegion.latitude,
-      longitude: initialRegion.longitude,
-      latitudeDelta: initialRegion.latitudeDelta,
-      longitudeDelta: initialRegion.longitudeDelta
+  //
+  const customizationCreation = (marker: MapMarker) => {
+    const match = customizers.find(({ key }) => key === marker.key)
+    if (match) {
+      return match
+    } else {
+      return {
+        color: "gray",
+        circleFillColor: "rgba(0, 0, 0, 0.5)",
+        circleStrokeColor: "gray",
+        circleStrokeWidth: 1
+      }
+    }
+  }
+
+  // When clicking on the marker, zoom towards where it is.
+  function onMarkerClick (lat: number, long: number) {
```

Comment:
It looks like the purpose of this function is to center the map on the marker; we can rename the function to something like "centerMapOnMarker()" for clarity.